### PR TITLE
Cloud 1457

### PR DESCRIPTION
**Pull request by DKagan07**
Adding logic to prevent creation of database names and table names

### CODE REVIEW

File: dax/controller/sqldb/schemar.go [hunk 1]
```diff
@@ -51,12 +51,19 @@ func (s *Schemar) CreateDatabase(tx dax.Transaction, qdb *dax.QualifiedDatabase)
 	}
 
 	org := &models.Organization{ID: string(qdb.OrganizationID)}
-	if ok, err := dt.C.Where("id = ?", qdb.OrganizationID).Exists(org); err != nil {
-		return errors.Wrap(err, "checking for org")
-	} else if !ok {
+	if err := dt.C.Eager().Where("id = ?", qdb.OrganizationID).First(org); isNoRowsError(err) {
```
Comment (travisturner) [inline dax/controller/sqldb/schemar.go:54]: I don't understand the change to the org query here. Was it not working before?

**Review by travisturner — CHANGES_REQUESTED** [review]

**Review by jaffee — APPROVED** [review]

File: dax/controller/sqldb/schemar.go [hunk 2]
```diff
@@ -244,6 +252,13 @@ func (s *Schemar) CreateTable(tx dax.Transaction, qtbl *dax.QualifiedTable) erro
 		return dax.NewErrInvalidTransaction("*sqldb.DaxTransaction")
 	}
 
+	// Check if TableID returns an error, that table does not exist and can be created
```
Comment (travisturner) [inline dax/controller/sqldb/schemar.go:255]: This comment doesn't seem to apply to what's happening.

File: dax/errors.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ const (
 	ErrDatabaseIDExists         errors.Code = "DatabaseIDExists"
 	ErrDatabaseIDDoesNotExist   errors.Code = "DatabaseIDDoesNotExist"
 	ErrDatabaseNameDoesNotExist errors.Code = "DatabaseNameDoesNotExist"
+	ErrDatabaseNameExists       errors.Code = "DatabaseNameExists"
```
Comment (travisturner) [inline dax/errors.go:15]: Is this one used?
Maybe we should move both of the new error types here to the `dax` package (instead of `dax/schemar`).

**Review by travisturner — COMMENTED** [review]
one command and some suggested error package move.

**Review by travisturner — APPROVED** [review]

**Pull request merged**